packages/php-wasm/universal/src/lib/base-php.ts

```diff
@@ -292,6 +292,12 @@ export abstract class BasePHP implements IsomorphicLocalPHP {
 				throw error;
 			}
 			return response;
+		} catch (e) {
+			this.dispatchEvent({
```

Comment:
Cool!

Reply:
Would it be useful to also emit response information when available?

Reply:
> Note this might either signify a fatal error in PHP, if request.throwOnError is true, or a more serious issue like, e.g., Asyncify exception.

Thanks! Both are valid cases for returning the event, but it would be good to distinguish between them.

Reply:
> Also, I wonder whether we should remove throwOnError and always act as if it was set to true – that's probably out of scope for this PR.

Let's take a look at this separately. I'm trying to avoid changing current logs in this project to stay focused.
But there is definitely room for improvement in how we throw and display errors.

Reply:
> Would it be useful to also emit response information when available? 

Do you mean emit the response when the request is successful? 
I don't know how I would do it for failed requests. There is `#getResponseHeaders` for getting the headers, but I'm not sure how useful that is.

Reply:
> Thanks! Both are valid cases for returning the event, but it would be good to distinguish between them.

Actually, I'm having second thoughts about this. A PHP fatal error would be triggered by `<?php throw new Exception();` and is harmless, whereas Asyncify-related crash could trash the entire PHP runtime. They feel like issues of completely different weight that should be handled differently, e.g. one could be perhaps be expected and ignored in some cases, and the other would require an expedited fix in Playground core.

Reply:
PHP exceptions won't trigger this. Only internal errors like Asyncify. 
For some reason, I thought they would, but after testing it, it doesn't.
You get a WordPress error screen and can read the log in the browser console/debug.log.

Reply:
I'm unable to recreate this. Do you have an example?

I tried throwing an error in 0-playground.php and introducing syntax errors. Both just returned a 500, but didn't trigger the event.
![Screenshot from 2024-03-22 10-50-35](https://github.com/WordPress/wordpress-playground/assets/1199991/b9b758cd-8436-4b01-aa7a-4321129b85a0)

Reply:
```ts
	php.addEventListener('request.error', (e) => {
		console.log("Request error");
	});
	php.run({
		throwOnError: true,
		code: `<?php
		syntax_error
		`,
	});
```

Reply:
Thanks! I added a[ flag that will prevent the modal from triggering](https://github.com/WordPress/wordpress-playground/pull/1102/commits/978dec896c377f957f6a0f310d11919d6b0522ea).

There is still value in catching these errors in some cases. For example if a developer uses Playground to demo their product and wants to know when it breaks on Playground.

Playground.WordPress.net won't show the modal, but a developer might decide to log the message somewhere to improve their demo experience.